pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dto/user/UserDto.java

```diff
@@ -36,10 +37,20 @@ public Optional<String> getEmail() {
         return Optional.ofNullable(email);
     }
 
+    // TODO: getEmail should always throw on a missing email since it is not nullable
+    // but there are a lot of callers. Feel free to chase them all down and rename this method -DC
+    public String getEmailOrThrow() {
+        return getEmail().orElseThrow(() -> new DsmInternalError("User email cannot be null"));
```

Comment:
Its a DTO and we are not setting the email.
DTO should just return  whatever exists rather than exception and up to caller to decide what to do.

Reply:
Yup, perhaps that's the downside of using the same data structure for input and output, but in this case you basically expect a non-null email for both (not nullable in DB). Are you thinking that DTOs can be used more widely that just DB input and output?

I added this because we are repeating similar logic elsewhere in the code, and I was just adding a few more instances of the same code. So having a central place for the logic made sense. 

Are you concerned about having the call, or my comment that says we should not have the other call (that does not throw)?

Reply:
I was thinking more of like getEmail() and just return whatever was set into the Dto and let the callers handle nulls.
However since the method is getEmailOrThrow() , may be its ok to leave it this way.

Reply:
We already have a getEmail call, which returns an optional. That means all users are forced to process the return in some fashion. This is a convenience method that benefits the great majority of callers who are effectively asserting that the email value is not null.

I can remove this and go back to calling getEmail... it just ends up in a lot of redundant code in the callers.

Reply:
If we already have the getEmail call, lets leave the new convenience method in there. sorry for the confusion.